cmd/agent/install_mac_os.sh

```diff
@@ -54,6 +54,17 @@ if [ -n "$DD_SYSTEMDAEMON_INSTALL" ]; then
         printf "\033[31mDD_SYSTEMDAEMON_USER_GROUP can't contain '>' or '<', because it will be used in XML file\033[0m\n"
         exit 1;
     fi
+    systemdaemon_user="$(echo "$systemdaemon_user_group" | awk -F: '{ print $1 }')"
+    systemdaemon_group="$(echo "$systemdaemon_user_group" | awk -F: '{ print $2 }')"
+    if ! id -u "$systemdaemon_user" >/dev/null 2>&1 ; then
+        printf "\033[31mUser $systemdaemon_user not found, can't proceed with installation\033[0m\n"
+        exit 1;
+    fi
+    # groups output is in form `username : group1 group2 ...`, so we use `cut` to strip the `username :` part
+    if ! groups "$systemdaemon_user" | cut -d' ' -f3- | grep -w "$systemdaemon_group" >/dev/null 2>&1; then
```

Comment:
With this check, we're requiring the user to be part of the group, which isn't mandatory in the Linux file permissions model. Is this on purpose?

Reply:
I just tried and it does seem like a completely arbitrary group can be provided. I'll change this.

Reply:
Done.